optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/DroolsVariableFactory.java

```diff
@@ -37,28 +42,106 @@
+     * As defined by {@link #createVariable(String, Variable, Variable, Variable, TriFunction, boolean)}
+     * with no flattening.
+     */
+    default <U, V, W, Result_> Variable<Result_> createVariable(String baseName, Variable<U> source1, Variable<V> source2,
+            Variable<W> source3, TriFunction<U, V, W, Result_> mapping) {
+        return createVariable(baseName, source1, source2, source3, mapping, false);
+    }
+
+    /**
+     * As defined by {@link #createVariable(String, Variable, Variable, Variable, Variable, QuadFunction, boolean)}
+     * with no flattening.
+     */
+    default <U, V, W, Y, Result_> Variable<Result_> createVariable(String baseName, Variable<U> source1, Variable<V> source2,
+            Variable<W> source3, Variable<Y> source4, QuadFunction<U, V, W, Y, Result_> mapping) {
+        return createVariable(baseName, source1, source2, source3, source4, mapping, false);
+    }
 
     /**
-     * Declare a new {@link Variable} with a given name, the contents of which will be identical to the source variable.
-     * Delegates to {@link DSL#declarationOf(Class, String, org.drools.model.DeclarationSource)}.
+     * Declares a new {@link Object}-typed variable, see {@link #createVariable(Class, String)} for details.
+     */
+    default <U> Variable<U> createVariable(String baseName) {
+        return (Variable<U>) createVariable(Object.class, baseName);
+    }
+
+    /**
+     * Declare a new {@link Variable} with a given name,
+     * where the value of the variable will be read from the provided source variable.
+     * If the value is {@link Iterable},
+     * the final argument to this method controls whether the elements will be treated individually or not.
      *
      * @param baseName name of the variable, mostly useful for debugging purposes. Will be decorated by a numeric
      *        identifier to prevent multiple variables of the same name to exist within left-hand side of a single rule.
-     * @param source source of the variable
-     * @param <X> generic type of the variable
-     * @return new variable declaration, not yet bound to anything
+     * @param source the variable the value of which will be set to the new variable
+     * @param flatten if true, we will flatten the collection and return one element after another
+     * @param <U> generic type of the variable
+     * @return never null
      */
-    <X> Variable<X> createVariable(String baseName, Variable<X> source);
+    <U> Variable<U> createVariable(String baseName, Variable<U> source, boolean flatten);
 
     /**
-     * Declares a new {@link Object}-typed variable, see {@link #createVariable(Class, String)} for details.
+     * Declare a new {@link Variable} with a given name,
+     * where the value of the variable will be computed from the provided source variable.
+     * If the computed value is {@link Iterable},
+     * the final argument to this method controls whether the elements will be treated individually or not.
```

Comment:
These lines duplicate the javadoc starting on line 98. Could we use a @link like in the others?

Reply:
The two are not exactly duplicates, the verbs are different. Therefore I'd rather keep it as is.